ably/auth.go

```diff
@@ -42,21 +63,87 @@ func (c Capability) String() string {
 }
 
 type Token struct {
-	Token      string     `json:"token"`
-	KeyName    string     `json:"keyName"`
-	Expires    int64      `json:"expires"`
-	Issued     int64      `json:"issued"`
-	Capability Capability `json:"capability"`
+	Token      string     `json:"token" msgpack:"token"`
+	KeyName    string     `json:"keyName" msgpack:"keyName"`
+	Expires    int64      `json:"expires" msgpack:"expires"`
+	Issued     int64      `json:"issued" msgpack:"issued"`
+	Capability Capability `json:"capability" msgpack:"capability"`
+}
+
+var _ msgpack.Unmarshaler = (*Token)(nil)
+
+func (tok *Token) UnmarshalMsgpack(p []byte) error {
+	// This method is workaround for msgpack decoder, which tries to decode fields
+	// using its original type and (finding field byte boundary) and then passes
+	// the bytes to custom unmarshaller.
+	//
+	// Decoding *Token directly fails with:
+	//
+	//   msgpack: invalid code ab decoding map length
+	//
```

Comment:
@lmars I've already rewritten this getting rid of the marshal / unmarshal dance - https://github.com/rjeczalik/ably-go/commit/247486357d1a8011d1347306c226aed51f607f4a#diff-fe17adcb123ff94418b17414c9d8940dR14. The capabilities are kept as encoded string with method helper `(*TokenRequest).Capability() Capability` - similarly to how `(url.URL).RawQuery` + `(url.URL).Query() url.Values` works.

I'm going to send the rewrite in separate PR. With fix for the flaky test.